src/ClientResolver.php

```diff
@@ -412,6 +427,12 @@ public static function _apply_endpoint_provider(callable $value, array &$args)
             if (isset($result['signatureVersion'])) {
                 $args['config']['signature_version'] = $result['signatureVersion'];
             }
+            if (isset($result['signingRegion'])) {
+                $args['config']['signing_region'] = $result['signingRegion'];
```

Comment:
Is the user provides an explicit signing region in the client constructor, it should override what comes back from the endpoint provider. Is that case handled?